### PR TITLE
Implements basic discard/keep editing functionality

### LINKED ISSUE #5940: Aztec: Add Discard / Save Draft / Keep Editing action sheet when closing an unsaved post
Add the Discard / Save Draft / Keep Editing action sheet that comes up in the hybrid editor when closing an unsaved post.

The logic should be the same we had.

### PR DESCRIPTION
**Pull request by astralbodies**
Closes #5940 

## Testing
**New Posts**
1. Turn on the native editor.
2. Compose a new post.
3. Cancel editing after: no edits, just a title, just a body, both title and body.
4. No edits should result in an immediate cancel. The latter three should present the option to discard.
5. After discarding the post should not appear in either published or drafts in the post list.

**Existing Posts**
1. Edit an existing post.
2. Make no edits and cancel. No discard option should present itself.
3. Make an edit and cancel. Discard option should present.

## Notes
1. A follow-up PR for develop will refactor PostService to remove the class methods for main context creation.
2. I will also follow up with pulling the revision creation logic out of the VCs everywhere and place that into the PostService. No business logic like this should be in any VC.
3. There are some TODOs in place (which pains me) but facilitates small PRs for this initial push.

Needs review: @diegoreymendez

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift [hunk 1]
```diff
@@ -319,7 +336,7 @@ extension AztecPostViewController
     }
 }
 
-
+// MARK: -
```
Comment (diegoreymendez) [inline WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift]: Is this intentional?

When looking at the complete file in Xcode it feels a bit arbitrary.

**Comment by diegoreymendez** [comment]
Here's what I found out testing on the iPad 2 Simulator:

**New posts:**
- [ 👍] No  edits resulted in an immediate cancel.
- [ 👍] Any other scenario showed the option to discard changes

**Existing posts:**
- [ 👍 ] No edits resulted in the post closing as expected.
- [ 👍 ] Editing the title triggered the discard / save draft dialog as expected.
- [ 👎 ] Editing the body didn't trigger the discard / save draft dialog.

**Comment by astralbodies** [comment]
@diegoreymendez I'm not able to reproduce the discard dialog not coming up. What type of edit were you doing in the body?

**Comment by diegoreymendez** [comment]
@astralbodies - Here's a movie with what I'm seeing.  Not sure what the difference may be, so let me know if you need anything else.

https://cloudup.com/cACE2cYseqn

**Comment by diegoreymendez** [comment]
@astralbodies - Feel free to merge and open up a new issue if necessary. :shipit:

**Pull request merged**